interfaces/builtin/content.go

```diff
@@ -169,23 +168,12 @@ func (iface *contentInterface) path(attrs interfaces.Attrer, name string) []stri
 // $SNAP_COMMON. If there are no variables then $SNAP is implicitly assumed
 // (this is the behavior that was used before the variables were supporter).
 func resolveSpecialVariable(path string, snapInfo *snap.Info) string {
-	if strings.HasPrefix(path, "$SNAP/") || path == "$SNAP" {
-		// NOTE: We use dirs.CoreSnapMountDir here as the path used will be always
-		// inside the mount namespace snap-confine creates and there we will
-		// always have a /snap directory available regardless if the system
-		// we're running on supports this or not.
-		// TODO parallel-install: use of proper instance/store name
-		return strings.Replace(path, "$SNAP", filepath.Join(dirs.CoreSnapMountDir, snapInfo.InstanceName(), snapInfo.Revision.String()), 1)
-	}
-	if strings.HasPrefix(path, "$SNAP_DATA/") || path == "$SNAP_DATA" {
-		return strings.Replace(path, "$SNAP_DATA", snapInfo.DataDir(), 1)
-	}
-	if strings.HasPrefix(path, "$SNAP_COMMON/") || path == "$SNAP_COMMON" {
-		return strings.Replace(path, "$SNAP_COMMON", snapInfo.CommonDataDir(), 1)
+	// NOTE: the variables are expanded in the context of the snap's mount namespace
+	if strings.HasPrefix(path, "$") {
+		return snapInfo.ExpandSnapMountVariables(path)
 	}
 	// NOTE: assume $SNAP by default if nothing else is provided, for compatibility
-	// TODO parallel-install: use of proper instance/store name
-	return filepath.Join(filepath.Join(dirs.CoreSnapMountDir, snapInfo.InstanceName(), snapInfo.Revision.String()), path)
+	return filepath.Join(snapInfo.ExpandSnapMountVariables("$SNAP"), path)
```

Comment:
Uh oh... That's completely different from what we had here before. The new logic means "$PRUNE/foo" becomes /foo, and people can now content-mount wherever they want.

The whole function needs to be restored, and the logic tuned as appropriate.

Reply:
On a side note `ExpandSnapMountVariables()` should probably leave unknown variables untouched, so that `$PRUNE/foo` expands to `$PRUNE/foo` and not `/foo`.

Reply:
That means unexpected behavior when we hand those variables for expansion elsewhere. We should either expand, or alternatively error out on unknowns.

Reply:
Ok, let's error out then. I'll push this as a separate PR.

This will get updated to:
```
func (s *Info) ExpandSnapVariables(path string) (string, error)
```
Since we're also sometimes calling `foo.ExpandSnapVariables("$SNAP")` explicitly, I'll add a `MustExpandSnapVariables()` variant which panics on errors. Sounds good?